### ISSUE TITLE
Implement boolean search

### ISSUE THREAD

**Issue by jasonwebber-bl** [ukwa/ukwa-ui#149]
AND, OR and NOT functions should work in search.

**Comment by anjackson** [comment]
As I understand it, the query parser we are using (as per #71) should support AND OR and NOT and also the syntax described [here](https://lucene.apache.org/solr/guide/6_6/the-dismax-query-parser.html#TheDisMaxQueryParser-TheqParameter).

**Comment by crarugal** [comment]
Works well when "s" is added to "doctor" term, seems like search terms work only if they are exact:

"haphard NOT doctors" -------works unlike "haphazard NOT doctor"
![image](https://user-images.githubusercontent.com/18530934/45815462-a60eb000-bcd0-11e8-8250-e766345f0c58.png)
https://dev.webarchive.org.uk/en/ukwa/search?view_filter=va&content_type=Web+Page&from_date=&to_date=&modal_filter_domains_vals=&modal_filter_suffix_vals=&modal_filter_documenttypes_vals=&modal_filter_collections_vals=&filter_array_x=&filter_array_x_item=&search_location=full_text&text=haphazard+NOT+doctors&view_sort=nto&view_count=50

**Comment by min2ha** [comment]
@crarugal lets be more accurate - if you put not "doctor" but "doctors" instead search results will be different.

**Comment by crarugal** [comment]
@min2ha yep I agree, just wanted to note that the most relevant search for "haphazard NOT doctor" had doctors at the top, whereas results further down had no "doctors". I just found it a bit unusual that "**haphazard, doctor**s" was highlighted.

Top results for "haphazard NOT doctor"
![image](https://user-images.githubusercontent.com/18530934/45816062-78c30180-bcd2-11e8-82b9-dab6f66fa730.png)

**Comment by crarugal** [comment]
Boolean seems to work, but I think search in general will need further tuning which could be addressed in other issues. Also I'm not sure if AND is being tokenized, given the large discrepancy in the number of results in the searches performed below.


AND results for "tony AND blaire",

![image](https://user-images.githubusercontent.com/18530934/45818328-48329600-bcd9-11e8-94b0-c0849ab999b3.png)


https://dev.webarchive.org.uk/en/ukwa/search?view_filter=vool&content_type=Web+Page&from_date=&to_date=&modal_filter_domains_vals=&modal_filter_suffix_vals=&modal_filter_documenttypes_vals=&modal_filter_collections_vals=&filter_array_x=&filter_array_x_item=&search_location=full_text&text=tony+AND+blaire&view_sort=nto&view_count=50

Far more results for "tony blaire"

![image](https://user-images.githubusercontent.com/18530934/45819933-7e721480-bcdd-11e8-9fb2-c75864063d28.png)

https://dev.webarchive.org.uk/en/ukwa/search?text=tony+AND+blaire&search_location=full_text&reset_filters=false&content_type=Web+Page

Is this a stopwords issue?

**Comment by crarugal** [comment]
OR also works but again I'm not sure if OR is being tokenized

![image](https://user-images.githubusercontent.com/18530934/45822231-e9721a00-bce2-11e8-9347-789c212a6113.png)

Looks like "or" and "un" have been emboldened; not sure if this is a search tokenizing issue or if it's just the front-end emboldening results that match search terms. 
![image](https://user-images.githubusercontent.com/18530934/45822425-58e80980-bce3-11e8-866a-595037c682cf.png)

**Comment by crarugal** [comment]
As expected, more results are returned when using OR

![image](https://user-images.githubusercontent.com/18530934/45823150-fabc2600-bce4-11e8-92c2-a750284936d9.png)

**Comment by crarugal** [comment]
I think this could be closed, with the caveat that these minor issues could be returned to when tuning search in later milestones @jasonwebber-bl @nicolabingham @anjackson @min2ha
